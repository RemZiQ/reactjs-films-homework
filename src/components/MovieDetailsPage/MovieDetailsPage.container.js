import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { currentMovie } from '../../modules/module_Search/selectors';
import MovieDetailsPage from './MovieDetailsPage';

const mapStateToProps = createStructuredSelector({ currentMovie });

export default connect(mapStateToProps, null)(MovieDetailsPage);
