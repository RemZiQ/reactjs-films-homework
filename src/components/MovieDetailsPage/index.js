export { default } from './MovieDetailsPage.container';
