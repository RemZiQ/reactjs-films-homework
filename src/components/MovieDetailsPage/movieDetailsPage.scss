.pageHeader {
  display: flex;
  box-sizing: border-box;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  max-width: 2400px;
  min-height: 935px;
  padding: 30px 20px 50px;
  background-size: cover !important;
}
