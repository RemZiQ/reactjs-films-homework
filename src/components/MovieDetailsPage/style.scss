.pageHeader{
  display: flex;
  box-sizing: border-box;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  max-width: 2400px;
  min-height: 1080px;
  max-height: 1100px;
  padding: 30px 20px 50px;
  background: linear-gradient(to bottom, rgba(0, 0, 0, 0.5) 0%, transparent 30%), 
    linear-gradient(to top, rgba(0, 0, 0, 0.8) 0%, transparent 30%), 
    url(../../image/junglebook.png) no-repeat black;
}