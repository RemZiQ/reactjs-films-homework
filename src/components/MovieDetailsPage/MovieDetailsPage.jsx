import React from 'react';
import { connect } from 'react-redux';
import propTypes from 'prop-types';


import MovieDetailsPageNav from '../MovieDetailsPageNav';
import MovieDetailsPageFooter from '../MovieDetailsPageFooter';
import './movieDetailsPage.scss';


export const MovieDetailsPage = ({ currentMovie }) => {
  const urlImage = 'https://image.tmdb.org/t/p/original';
  if (currentMovie) {
    const style = {
      background: `linear-gradient(to bottom, rgba(0, 0, 0, 0.5) 0%, transparent 30%),
      linear-gradient(to top, rgba(0, 0, 0, 0.8) 0%, transparent 30%),
      url(${urlImage}${currentMovie.backdrop_path}) no-repeat black`,
    };
    return (
      <header
        className="pageHeader"
        style={style}
        id="pageHeaderID"
      >
        <MovieDetailsPageNav />
        <MovieDetailsPageFooter />
      </header>
    );
  }
  return <div>loading...</div>;
};

const mapStateToProps = store => ({ currentMovie: store.currentMovie });

MovieDetailsPage.propTypes = {
  currentMovie: propTypes.objectOf(propTypes.any).isRequired,
};

export default connect(mapStateToProps, null)(MovieDetailsPage);
