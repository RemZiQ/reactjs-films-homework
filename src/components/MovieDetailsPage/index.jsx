import React from 'react';
import MovieDetailsPageNav from '../MovieDetailsPage__Nav/index';
import MovieDetailsPageFooter from '../MovieDetailsPage__Footer/index';
import './style.scss';


const  MovieDetailsPage = () => {
    return (
      <header className="pageHeader" id="pageHeaderID">
      <MovieDetailsPageNav></MovieDetailsPageNav>
      <MovieDetailsPageFooter></MovieDetailsPageFooter>
      </header>
    )
}

export default MovieDetailsPage;