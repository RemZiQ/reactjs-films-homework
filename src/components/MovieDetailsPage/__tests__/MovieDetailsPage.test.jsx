import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MovieDetailsPage from '../MovieDetailsPage';


it('MovieDetailsPage renders well without props', () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(<MovieDetailsPage />);
  expect(result).toMatchSnapshot();
});

it('MovieDetailsPage renders well with props', () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(<MovieDetailsPage currentMovie={{ backdrop_path: 'abc' }} />);
  expect(result).toMatchSnapshot();
});
