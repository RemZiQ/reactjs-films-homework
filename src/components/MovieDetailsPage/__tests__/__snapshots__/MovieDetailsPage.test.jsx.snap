// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsPage renders well with props 1`] = `
<header
  className="pageHeader"
  id="pageHeaderID"
  style={
    Object {
      "background": "linear-gradient(to bottom, rgba(0, 0, 0, 0.5) 0%, transparent 30%),
      linear-gradient(to top, rgba(0, 0, 0, 0.8) 0%, transparent 30%),
      url(https://image.tmdb.org/t/p/originalabc) no-repeat black",
    }
  }
>
  <withRouter(Connect(Nav)) />
  <Connect(MovieDetailsPageFooter) />
</header>
`;

exports[`MovieDetailsPage renders well without props 1`] = `
<div>
  loading...
</div>
`;
